spring-integration-core/src/main/java/org/springframework/integration/endpoint/SourcePollingChannelAdapter.java

```diff
@@ -129,6 +133,13 @@ protected boolean isReceiveOnlyAdvice(Advice advice) {
 	@Override
 	protected void applyReceiveOnlyAdviceChain(Collection<Advice> chain) {
 		if (AopUtils.isAopProxy(this.source)) {
+			Advisor[] advisors = ((Advised) this.source).getAdvisors();
+			for (Advisor advisor : advisors)  {
+				Advice advice = advisor.getAdvice();
+				if (advice != null && this.appliedAdvices.contains(advice)) {
+					((Advised) this.source).removeAdvice(advice);
```

Comment:
Looks like, when we come to the `Advisor` solution, we only can iterate over `applied` and call `Advised.removeAdvisor(Advisor);`
Although even right now we can use `Advised.removeAdvise()`...